### PR TITLE
:warning: Switch to 1 secret per certificate

### LINKED ISSUE #135: Modify cert secret to be 1 secret per cert
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
In  https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/pull/126 certificate generation and secret lookup was added to the bootstrap controller. We had a discussion on the secret format related to cert-manager. In order to support cert-manager we will need 1 secret per cert. 

See details here: https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/pull/126#discussion_r314039485

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by ncdc**
**What this PR does / why we need it**:
Switch to 1 secret per certificate, instead of a single certificate.

Supersedes & replaces #153

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #135

### CODE REVIEW

**Comment by ncdc** [comment]
@moshloop wrote in his PR:
> I also want to make specifying the etcd,proxy and sa certs optional

Do you all want me to do that here?

File: certs/certs.go [hunk 1]
```diff
@@ -30,8 +30,18 @@ import (
 )
 
 const (
-	rsaKeySize   = 2048
 	duration365d = time.Hour * 24 * 365
+	rsaKeySize   = 2048
+	tlsKey       = "tls.key"
+	tlsCrt       = "tls.crt"
```
Comment (vincepri) [inline certs/certs.go]: The above constants are also defined upstream, if you want to point at those instead

File: go.sum [hunk 2]
```diff
@@ -251,12 +236,13 @@ k8s.io/klog v0.4.0 h1:lCJCxf/LIowc2IGS9TPjWDyXY4nOmdGdfcwwDQCOURQ=
...
-k8s.io/utils v0.0.0-20190506122338-8fab8cb257d5 h1:VBM/0P5TWxwk+Nw6Z+lAw3DKgO76g90ETOiA6rfLV1Y=
 k8s.io/utils v0.0.0-20190506122338-8fab8cb257d5/go.mod h1:sZAwmy6armz5eXlNoLmJcl4F1QuKu7sr+mFQ0byX7Ew=
 k8s.io/utils v0.0.0-20190809000727-6c36bc71fc4a h1:uy5HAgt4Ha5rEMbhZA+aM1j2cq5LmR6LQ71EYC2sVH4=
 k8s.io/utils v0.0.0-20190809000727-6c36bc71fc4a/go.mod h1:sZAwmy6armz5eXlNoLmJcl4F1QuKu7sr+mFQ0byX7Ew=
-sigs.k8s.io/cluster-api v0.0.0-20190829154355-58b8460b80c5 h1:iF/85F2lFHJLDxJk2KQC1Yps51bAHCpehNryTvSr1vM=
-sigs.k8s.io/cluster-api v0.0.0-20190829154355-58b8460b80c5/go.mod h1:5ODrAYsZYC0gHyqu20VcqnD3uLXaqzFN9HzBwYAAKNs=
+sigs.k8s.io/cluster-api v0.0.0-20190829222858-82b282b06aaf h1:YkMawPF8j53aoMqpDFdaCXd0jYX4nOm/33677WLUfWQ=
+sigs.k8s.io/cluster-api v0.0.0-20190829222858-82b282b06aaf/go.mod h1:5ODrAYsZYC0gHyqu20VcqnD3uLXaqzFN9HzBwYAAKNs=
+sigs.k8s.io/cluster-api v0.0.0-20190830155554-d495192f76e0 h1:qV0Rvi06IYwPwlmvCt8F2TTlp62Cn7hZCmI1CTtymFM=
+sigs.k8s.io/cluster-api v0.0.0-20190830155554-d495192f76e0/go.mod h1:5ODrAYsZYC0gHyqu20VcqnD3uLXaqzFN9HzBwYAAKNs=
```
Comment (vincepri) [inline go.sum]: Did `go mod tidy` leave 2 versions of CAPI here?

File: certs/certs.go [hunk 3]
```diff
@@ -30,8 +30,18 @@ import (
...
+	//ClusterCAName is the secret name suffix for apiserver CA
```
Comment (vincepri) [inline certs/certs.go]: Leave a space after `//`?

**Comment by chuckha** [comment]
no additional comments just delete more code that the linter is complaining about, run go mod tidy and then lgtm

File: certs/certs.go (hunk 1 above)
Comment (ncdc) [reply certs/certs.go]: We should maybe remove them from CAPI and only have them here?

File: go.sum (hunk 2 above)
Comment (ncdc) [reply go.sum]: Fixed in latest push

File: certs/certs.go (hunk 1 above)
Comment (ncdc) [reply certs/certs.go]: I'm going to use the ones in this repo for the time being

Comment (vincepri) [reply certs/certs.go]: They're currently used in CAPI

Comment (ncdc) [reply certs/certs.go]: They're showing up unused in my editor (at least nothing in CAPI references them)

Comment (detiber) [reply certs/certs.go]: I would expect CAPI to use tlsCrt with the kubeconfig generation...  it wouldn't need tlsKey

Comment (vincepri) [reply certs/certs.go]: Are you on latest master? The first two are used to create the certs, the TLSkey/crt are used to retrieve the Cluster CA

Comment (ncdc) [reply certs/certs.go]: I see them in use now that I pulled

**Pull request merged**